src/main/clm_driver.F90

```diff
@@ -1029,6 +1038,14 @@ subroutine clm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate, ro
             frictionvel_inst, photosyns_inst, drydepvel_inst)
        call t_stopf('depvel')
 
+       if (use_cropcal_streams .and. is_beg_curr_year()) then
+          ! ============================================================================
+          ! Update crop calendars
+          ! ============================================================================
+          call cropcal_interp(bounds_clump, filter_inactive_and_active(nc)%num_pcropp, filter_inactive_and_active(nc)%pcropp, crop_inst)
```

Comment:
Yeah, using `filter_inactive_and_active` did solve an issue I had where prescribed sowing dates and cultivars were Ignored the first year after a crop patch became active. E.g., this behavior:
```
Lon 297.5 lat 8.5263157894736, irrigated_sugarcane
      gs  SDATES      GDDHARV  HDATES  HARVEST_REASON
0   1961     NaN          NaN     NaN             NaN
...
19  1980     NaN          NaN     NaN             NaN
20  1981     1.0  1850.000000   184.0             1.0
21  1982     1.0  2821.125977   226.0             1.0
22  1983     1.0  2821.125977   232.0             1.0
```

Reply:
Okay, thanks for the response on filter_inactive_and_active. I can't say I understand why that's the case, since the filters should be updated for the newly-active crops by this point in the driver loop, but I guess there's some subtlety here that I'm missing. Anyway, it seems fine to use the inactive_and_active filters here, so fine to leave as is.